src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,511 @@
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.BatchSize));
+            }
+        }
+
+        /// <summary>
+        /// The current running user
+        /// </summary>
+        public string Implementation
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Implementation));
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia code directory.
+        /// </summary>
+        protected string NvidiaDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, "mlperf", "NVIDIA");
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Pytorch code directory.
+        /// </summary>
+        protected string ExecutionDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.NvidiaDirectoryPath, this.Model, "implementations", this.Implementation);
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia data directory.
+        /// </summary>
+        protected string DataDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, this.Model);
+            }
+        }
+
+        /// <summary>
+        /// The output directory of MLPerf.
+        /// </summary>
+        protected string OutputDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.ExecutionDirectoryPath, "results");
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MLPerf Training workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.InitializationStarted", telemetryContext);
+
+            this.ThrowIfPlatformNotSupported();
+
+            await this.ThrowIfUnixDistroNotSupportedAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            MLPerfState state = await this.stateManager.GetStateAsync<MLPerfState>($"{nameof(MLPerfState)}", cancellationToken)
+                ?? new MLPerfState();
+
+            if (!state.Initialized)
+            {
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+                await this.ExecuteCommandAsync("newgrp", $"docker", this.ExecutionDirectoryPath, cancellationToken);
+
+                // If GPUConfig is not included in the MLPerf code but is supported
+                this.ReplaceGPUConfigFilesToSupportAdditionalGPUs();
```

Comment:
Considering future GPUs, like H100, it would be better to keep this replacement based on a flag, like customSystemRequired. Reason being, this file will be updated in repo of MLPerf with every new HW and we will be copying the older file which will not have support for newer GPU(H100 in current scenario). So, we can keep it optional so that with every new HW, we wont have to change this file and we will need to change this file only if any special custom system needs to be tested. This change is also required in MLPerf inference, i will raising the PR for that also.

Reply:
As of now I am not using any separate config files
The workload itself provides config files for GPUs, I have been using that.